### ISSUE TITLE
Getting `Pay::InvalidPaymentMethod`

### ISSUE THREAD

**Issue by archonic** [pay-rails/pay#381]
I updated Pay, hit an error, then backdated to a version I know passed tests before. I'm not sure how but I'm still getting `Pay::InvalidPaymentMethod` on the majority of my rspec tests. I didn't update the API version either. This is what my factory for Billable looks like:

```
factory :account do
  name { Faker::Movies::LordOfTheRings.location }
  subdomain { name.downcase.parameterize }
  owner_au_attributes { attributes_for(:accounts_user, :admin, user_attributes: attributes_for(:user)) }
  association :plan
  card_last4 { nil }
  card_type { nil }
  card_exp_month { nil }
  card_exp_year { nil }
  trial_ends_at { TRIAL_PERIOD_DAYS.days.from_now }
  initialize_with { Account.where(subdomain: subdomain).first_or_initialize }

  after(:create) do |account|
    account.processor = "stripe"
    account.subscribe(name: account.product.name, plan: account.plan.stripe_id)
  end
end
```

The default account factory is on a new trial. Calling `subscribe` without a payment method worked for many months before this point.

Do you have any recommendations for setting up a factory for a trialing account?

UPDATE: I've noticed that `Stripe.api_version = "2020-03-02"` in my stripe initializer seems to be getting ignored. Calls are coming through on 2020-08-27. How does Pay pin the API version?

**Comment by archonic** [comment]
I'm fairly certain that what I'm seeing is a result of an unintentional Stripe API version update, but it would appear that it exposed a Pay bug. If I'm calling `subscribe` on a trialing Billable model, I don't think it's meant to attempt payment.

**Comment by excid3** [comment]
Pay pins the version the same way. https://github.com/pay-rails/pay/blob/master/lib/pay/stripe.rb#L30

The `trial_ends_at` on Billable is for generic trials that aren't on a payment processor. It doesn't create a Pay::Subscription or anything. 

Calling `subscribe` will always create a subscription on the payment processor. You'd have to use their trial features if you want a trial on a subscription. Typically this will require a payment method, whereas a generic trial (via trial_ends_at on the Billable) does not.

**Comment by excid3** [comment]
This should be fixed by #385.

Problem was that Trial Subscriptions (without a card) were being validated for SCA (which isn't necessary since it's a trial).
